provisioning.go

```diff
@@ -0,0 +1,370 @@
+// clear the TPM via the physical presence interface.
+//
+// If mode is ProvisionModeClear, this function will attempt to clear the TPM before provisioning it. If owner clear has been disabled
+// (which will be the case if the TPM has previously been provisioned with this function), then ErrTPMClearRequiresPPI will be returned.
+// In this case, the TPM must be cleared via the physical presence interface by calling RequestTPMClearUsingPPI and performing a
+// system restart.
+//
+// This function will create and persist an endorsement key which requires knowledge of the authorization values for the storage
+// and endorsement hierarchies. If called with mode set to ProvisionModeClear, or if called just after clearing the TPM via the
+// physical presence interface, the authorization values for these hierarchies will be empty at the point that they are required.
+// If called with any other mode and if the authorization values have previously been set, they will need to be provided by calling
+// TPMConnection.EndorsementHandleContext().SetAuthValue() and TPMConnection.OwnerHandleContext().SetAuthValue() prior to calling
+// this function. If the wrong value is provided for either authorization, then a AuthFailError error will be returned. If the
+// correct authorization values are not known, then the only way to recover from this is to call the function with mode set to
+// ProvisionModeClear. If there is an object already stored at the location used for the endorsement key then this function will
+// evict it automatically from the TPM.
+//
+// This function will create and persist a storage root key, which requires knowledge of the authorization value for the storage
+// hierarchy. If called with mode set to ProvisionModeClear, or if called just after clearing the TPM via the physical presence
+// interface, the authorization value for the storage hierarchy will be empty at the point that it is required. If called with any
+// other mode and if the authorization value for the storage hierarchy has previously been set, it will need to be provided by calling
+// TPMConnection.OwnerHandleContext().SetAuthValue() prior to calling this function. If the wrong value is provided for the storage
+// hierarchy authorization, then a AuthFailError error will be returned. If the correct authorization value is not known, then the
+// only way to recover from this is to call the function with mode set to ProvisionModeClear. If there is an object already stored at
+// the location used for the storage root key then this function will evict it automatically from the TPM.
+//
+// This function will create a pair of NV indices used for locking access to sealed key objects, if necessary. These indices will be
+// created at 0x01801100 and 0x01801101. This requires knowledge of the authorization value for the storage hierarchy. If called with
```

Comment:
I suppose these are referred by hex notation because the user of the library never needs to specify them ?

Reply:
Yes, these are deliberately hardcoded so that they are the same on every device, and they aren't defined by any specs so there aren't any nice names for them.